src/frontend/src/lib/types/transaction.ts

```diff
@@ -28,3 +32,7 @@ export type TransactionUiCommon = Pick<Transaction, 'blockNumber' | 'from' | 'to
 	toExplorerUrl?: string;
 	fromExplorerUrl?: string;
 };
+
+export type UnifiedTransactionUi = (BtcTransactionUi | EthTransactionUi | IcTransactionUi) & {
+	transactionComponent: ComponentType;
```

Comment:
nitpick: should this be renamed to `component`?

Reply:
right, better